storage/localstore/mode_set.go

```diff
@@ -393,3 +399,31 @@ func (db *DB) setUnpin(batch *leveldb.Batch, addr chunk.Address) (err error) {
 
 	return nil
 }
+
+// setReUpload adds a pinned chunk to the push index so that it can be re-uploaded to the network
+func (db *DB) setReUpload(batch *leveldb.Batch, addr chunk.Address) (err error) {
+	item := addressToItem(addr)
+
+	// only pinned chunks should be re-uploaded
+	// this also prevents a race condition: a non-pinned chunk could be gc'd after it's put in the push index
+	// but before it is actually re-uploaded to the network
+	_, err = db.pinIndex.Get(item)
+	if err != nil {
+		return err
+	}
+
+	// put chunk item with retrieval data into the push index if not already present
+	retrievalDataIndexItem, err := db.retrievalDataIndex.Get(item)
+	if err != nil {
+		return err
+	}
+	itemPresent, err := db.pushIndex.Has(retrievalDataIndexItem)
+	if err != nil {
+		return err
+	}
+	if !itemPresent {
+		return db.pushIndex.PutInBatch(batch, retrievalDataIndexItem)
+	}
+
+	return nil
```

Comment:
I am suggesting this change, as put to the index is the happy path, and present is the exception. This is only a minor thing that is opinionated. This is of extremely low importance, feel free not to change.

```suggestion
	if itemPresent {
	    return nil
	}

	return db.pushIndex.PutInBatch(batch, retrievalDataIndexItem)
```

Reply:
i like this! thanks